packages/babel-helper-create-class-features-plugin/src/fields.ts

```diff
@@ -78,16 +89,26 @@ export function buildPrivateNamesNodes(
   return initNodes;
 }
 
+interface PrivateNameVisitorState {
+  privateNamesMap: PrivateNamesMap;
+  privateFieldsAsProperties: boolean;
+  redeclared?: string[];
+}
+
 // Traverses the class scope, handling private name references. If an inner
 // class redeclares the same private name, it will hand off traversal to the
 // restricted visitor (which doesn't traverse the inner class's inner scope).
-function privateNameVisitorFactory(visitor) {
-  const privateNameVisitor = {
+function privateNameVisitorFactory<S>(
+  visitor: Visitor<PrivateNameVisitorState & S>,
+) {
+  const privateNameVisitor: Visitor<PrivateNameVisitorState & S> = {
     ...visitor,
 
     Class(path) {
       const { privateNamesMap } = this;
-      const body = path.get("body.body");
+      // not using body.body since it will loose typing
+      const classBody = path.get("body");
+      const body = classBody.get("body");
```

Comment:
Actually, I tried this approach, which is not correct.

`NodePath<t.ClassBody["body"]>` resolved to `NodePath<(ClassMethod | ClassPrivateMethod | ClassPrivateProperty | ClassProperty | TSDeclareMethod | TSIndexSignature)[]>`, but what we need is `NodePath<ClassMethod | ClassPrivateMethod | ClassPrivateProperty | ClassProperty | TSDeclareMethod | TSIndexSignature>[]`

It's like `NodePath<Body[]>` to `NodePath<Body>[]`

Reply:
Oh ok. I'd prefer just `path.get("body").get("body")` then; we don't need the temp variable.

Reply:
Sounds great to me!

By the way, are there any chance that we could get same result type for `path.get("body.body")` and `path.get("body").get("body")`

Reply:
I can try writing `path.get`'s type definitions using TS template literal types, but I fear that TSC will complain because it causes a really big number of possible cases.